File: hackerrank/30 days of code/day 14, scope/scope.py
# https://www.hackerrank.com/challenges/30-scope/problem?isFullScreen=true

def main():
    input()  # Size not used
    numbers = list(map(int, input().split()))
    difference = Difference(numbers)
    difference.compute_difference()
    print(difference.maximum_difference)


class Difference:

    def __init__(self, elements):
        self.__elements = elements
        self.maximum_difference = 0

    def compute_difference(self):
        self.maximum_difference = max(self.__elements) - min(self.__elements)


if __name__ == '__main__':
    main()

File: hackerrank/30 days of code/day 14, scope/scope.cs
// https://www.hackerrank.com/challenges/30-scope/problem?isFullScreen=true

using static System.Console;

class Solution
{
    public static void Main()
    {
        ReadLine(); // Size not used
        List<int> numbers = [.. ReadLine().Split().Select(int.Parse)];
        var difference = new Difference(numbers);
        difference.ComputeDifference();
        WriteLine(difference.MaximumDifference);
    }
}

class Difference(List<int> elements)
{
    private readonly List<int> _elements = elements;
    public int MaximumDifference;

    public void ComputeDifference()
    {
        MaximumDifference = _elements.Max() - _elements.Min();
    }
}
